tests/test_pytypes.cpp

```diff
@@ -811,4 +813,11 @@ TEST_SUBMODULE(pytypes, m) {
         a >>= b;
         return a;
     });
+
+    m.def("annotate_tuple_float_str", [](const py::Tuple<py::float_, py::str> &) {});
+    m.def("annotate_tuple_empty", [](const py::Tuple<> &) {});
+    m.def("annotate_dict_str_int", [](const py::Dict<py::str, int> &) {});
+    m.def("annotate_list_int", [](const py::List<int> &) {});
+    m.def("annotate_set_str", [](const py::Set<std::string> &) {});
+    m.def("annotate_fn", [](const py::Callable<int(py::List<py::str>, py::str)> &) {});
```

Comment:
Might be good to add a couple more tests here to be double sure.

I would add 

`m.def("annotate_empty_fn", [](const py::Callable<void()> &) {});` and `m.def("annotate_nested_tuple", [](const py::Tuple<py::Tuple<<>> &) {});`


I think both of those edge cases should work, but good to double check.